Second Rust

GeorgeHahn opened the pull request:
The important commit here is `0f09539` - it switches to using configuration 0x01 and using interrupt reads instead of bulk reads.

GeorgeHahn committed:
Add Rust.gitignore

GeorgeHahn committed:
Add logging libs & switch to Windows-compatible libusb

GeorgeHahn committed:
Log everything

GeorgeHahn committed:
Fix USB: use config 1 & use `read_interrupt`

GeorgeHahn committed:
fixup: remove patch versions

GeorgeHahn committed:
Refactor functions to struct

BroderickCarlin commented:
This PR compiles/runs for me however it fails to claim the new interface specified. I am confused why our 2 hubs would have different interfaces exposed

GeorgeHahn committed:
Rudimentary command implementation

Based off of reversing work done here: https://hclxing.wordpress.com/2019/06/06/reverse-engineering-wyzesense-bridge-protocol-part-iii/
